compiler/executable/match_/planner/vertex/mod.rs

```diff
@@ -550,31 +559,32 @@ impl Costed for UnsatisfiablePlanner<'_> {
 }
 
 #[derive(Clone, Debug)]
-pub(super) struct NegationPlanner<'a> {
+pub(super) struct NestedNegationPlan<'a> {
     plan: ConjunctionPlan<'a>,
-    shared_variables: Vec<VariableVertexId>,
+    referenced_parent_vertex_ids: HashSet<VariableVertexId>,
 }
 
-impl<'a> NegationPlanner<'a> {
-    pub(super) fn new(plan: ConjunctionPlan<'a>, variable_index: &HashMap<Variable, VariableVertexId>) -> Self {
-        let shared_variables = plan.shared_variables().iter().map(|v| variable_index[v]).collect();
-        Self { plan, shared_variables }
+impl<'a> NestedNegationPlan<'a> {
+    pub(super) fn new(plan: ConjunctionPlan<'a>, parent_variable_index: &HashMap<Variable, VariableVertexId>) -> Self {
+        let referenced_parent_vertex_ids =
+            plan.referenced_input_variables().map(|v| parent_variable_index[&v]).collect();
+        Self { plan, referenced_parent_vertex_ids }
     }
 
-    fn is_valid(&self, ordered: &[VertexId], _graph: &Graph<'_>) -> bool {
-        self.variables().all(|var| ordered.contains(&VertexId::Variable(var)))
+    fn is_valid(&self, ordered: &[VertexId]) -> bool {
+        self.referenced_parent_vertex_ids.iter().all(|var| ordered.contains(&VertexId::Variable(*var)))
     }
 
-    pub(crate) fn variables(&self) -> impl Iterator<Item = VariableVertexId> + '_ {
-        self.shared_variables.iter().copied()
+    pub(crate) fn referenced_parent_vertex_ids(&self) -> impl Iterator<Item = VariableVertexId> + '_ {
+        self.referenced_parent_vertex_ids.iter().copied()
```

Comment:
I don't think we need to rename this honestly. The structs is called `*Planner` and the method is called `variables` because you're supposed to think about them the same way. This might be the more obvious name to someone who reads the `ir` package and passes through the compiler package but it's detrimental to someone who has to understand how the planner works.